replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/MetacardConfigLoader.java

```diff
@@ -54,97 +54,80 @@
+        () -> {
+          try {
+            security.runWithSubjectOrElevate(
+                () -> {
+                  getAllConfigs()
+                      .forEach(
+                          config -> {
+                            newConfigManager.save(config);
+                            removeConfig(config);
+                          });
+                  return null;
+                });
+          } catch (SecurityServiceException | InvocationTargetException e) {
+            LOGGER.warn(
+                "Error occured while trying to transform catalog saved metacards to persistent store saved configs. Legacy configs may not be accessible.",
+                e);
+          }
+          return null;
+        });
   }
 
-  public List<ReplicatorConfig> getAllConfigs() {
+  List<ReplicatorConfig> getAllConfigs() {
     return helper.getTypeForFilter(
         filterBuilder.attribute(Metacard.TAGS).is().like().text(ReplicationConfig.METACARD_TAG),
         this::getConfigFromMetacard);
   }
 
-  public void saveConfig(ReplicatorConfig replicationConfig) {
-    Metacard mcard = null;
-    if (replicationConfig.getId() != null) {
-      mcard = helper.getMetacardById(replicationConfig.getId());
-    }
-    boolean create = mcard == null;
-    mcard = getMetacardFromConfig(replicationConfig, mcard);
-
-    if (mcard == null) {
-      LOGGER.warn(
-          "Error creating/updating metacard for replication configuration {}",
-          replicationConfig.getName());
-      throw new ReplicationException(
-          "Error creating/updating metacard for replication configuration "
-              + replicationConfig.getName());
-    }
-
-    // we always save in the latest config version so
-    helper.setIfPresent(mcard, ReplicationConfig.VERSION, ReplicatorConfigImpl.CURRENT_VERSION);
-    if (create) { // create
-      try {
-        framework.create(new CreateRequestImpl(mcard));
-      } catch (IngestException | SourceUnavailableException e) {
-        LOGGER.warn(
-            "Error creating replication configuration for {}", replicationConfig.getName(), e);
-        throw new ReplicationException(
-            "Error creating replication configuration for " + replicationConfig.getName());
-      }
-    } else { // update
-      try {
-        framework.update(new UpdateRequestImpl(mcard.getId(), mcard));
-      } catch (IngestException | SourceUnavailableException e) {
-        LOGGER.warn(
-            "Error updating replication configuration for {}", replicationConfig.getName(), e);
-        throw new ReplicationException(
-            "Error updating replication configuration for " + replicationConfig.getName());
-      }
-    }
-  }
-
-  public void removeConfig(ReplicatorConfig replicationConfig) {
+  void removeConfig(ReplicatorConfig replicationConfig) {
     try {
       framework.delete(new DeleteRequestImpl(replicationConfig.getId()));
     } catch (IngestException | SourceUnavailableException e) {
```

Comment:
❓ Should we be handling `SourceUnavailableException` differently to let Failsafe above retry on these? Maybe a new `ReplicationSourceUnavailableException` or something similar extending `ReplicationException`

Reply:
I'm not sure what specifically causes an IngestException so I'm not sure whether it means we should try again. I think we should leave it for now and try again on either exception just to be safe. I don't think we need to log and rethrow the exception however

Reply:
One of the problem I see is that if you cannot delete it, than you will be continuously trying to re-upgrade it and potentially override what was already updated by users since the first time it was upgraded. Maybe, we should add a check to see if a particular object already exist in the new repo before attempting to override it.

`SourceUnavailableException` definitely sounds like one we should retry on.